backend/src/src-vulnerabilities/src/main/java/org/eclipse/sw360/vulnerabilities/VulnerabilityHandler.java

```diff
@@ -129,54 +123,57 @@ public VulnerabilityHandler() throws IOException, SW360Exception {
+                .filter(VulnerabilityHandler::releaseVulnerabilityRelationIsNotIncorrect)
                 .collect(Collectors.toList());
+        return getDtosFromRelations(filteredRelations, user);
     }
 
-    private List<VulnerabilityDTO> getVulsByReleaseId(String releaseId, User user) {
-        List<ReleaseVulnerabilityRelation> relations = dbHandler.getRelationsByReleaseIds(Arrays.asList(releaseId));
-        if (relations == null || relations.size() == 0) {
-            return Collections.emptyList();
-        }
-        List<VulnerabilityDTO> resultDtos = new ArrayList<>();
-        for (ReleaseVulnerabilityRelation relation : relations) {
-            resultDtos.add(getDtoFromRelation(relation, releaseId, user));
-        }
-        return resultDtos;
+    private List<ReleaseVulnerabilityRelation> getReleaseVulnerabilityRelationByReleaseId(Collection<String> releaseIds) {
+        List<ReleaseVulnerabilityRelation> relations = dbHandler.getRelationsByReleaseIds(releaseIds);
+        return MoreObjects.firstNonNull(relations, Collections.emptyList());
     }
 
-    private boolean releaseVulnerabilityRelationIsIncorrect(ReleaseVulnerabilityRelation relation) {
+    private static boolean releaseVulnerabilityRelationIsNotIncorrect(ReleaseVulnerabilityRelation relation) {
         if (!relation.isSetVerificationStateInfo()) {
-            return false;
+            return true;
         }
         List<VerificationStateInfo> stateHistory = relation.getVerificationStateInfo();
         VerificationStateInfo currentState = stateHistory.get(stateHistory.size() - 1);
-        return VerificationState.INCORRECT.equals(currentState.getVerificationState());
+        return !VerificationState.INCORRECT.equals(currentState.getVerificationState());
     }
 
-    private List<VulnerabilityDTO> getVulsByReleaseIdWithoutIncorrect(String releaseId, User user) {
-        List<ReleaseVulnerabilityRelation> relations = dbHandler.getRelationsByReleaseIds(Arrays.asList(releaseId));
-        if (relations == null || relations.size() == 0) {
-            return Collections.emptyList();
-        }
-        List<VulnerabilityDTO> resultDtos = new ArrayList<>();
-        for (ReleaseVulnerabilityRelation relation : relations) {
-            if (!releaseVulnerabilityRelationIsIncorrect(relation)) {
-                resultDtos.add(getDtoFromRelation(relation, releaseId, user));
-            }
-        }
-        return resultDtos;
+    private List<VulnerabilityDTO> getDtosFromRelations(List<ReleaseVulnerabilityRelation> relations, User user) {
+        Set<String> vulnerabilityIds = relations.stream()
+                .map(ReleaseVulnerabilityRelation::getVulnerabilityId)
+                .collect(Collectors.toSet());
+
+        List<Vulnerability> vulnerabilities = dbHandler.getByIds(Vulnerability.class, vulnerabilityIds);
+        Map<String, Vulnerability> vulnerabilityMap = vulnerabilities.stream()
```

Comment:
I think this does the same as `ThriftUtils.getIdMap()`

Reply:
Indeed. I changed this, and I took the liberty to make ThriftUtils more Java 8 compliant, I hope you don't mind.

Reply:
awesome!